Python/optimizer_analysis.c

```diff
@@ -482,15 +482,19 @@ optimize_uops(
     _PyUOpInstruction *corresponding_check_stack = NULL;
 
     _Py_uop_abstractcontext_init(ctx);
+
+    // Note: this must happen before frame_new, as it might override
+    // the result should frame_new set things to bottom.
+    ctx->done = false;
+    ctx->out_of_space = false;
+    ctx->contradiction = false;
```

Comment:
Would it make sense to move these inside `_Py_uop_abstractcontext_init`?

Reply:
Oh yeah good point!